src/main/groovy/nextflow/cloud/gce/pipelines/GooglePipelinesTaskHandler.groovy

```diff
@@ -0,0 +1,254 @@
+        events?.reverse()?.each {
+            log.trace "[GOOGLE PIPELINE] New event for task '$task.name' - time: ${it.get("timestamp")} - ${it.get("description")}"
+        }
+
+        if (operation.getDone()) {
+            log.debug "[GOOGLE PIPELINE] Task '$task.name' complete. Start Time: ${metadata?.getStartTime()} - End Time: ${metadata?.getEndTime()}"
+
+            // finalize the task
+            Integer xs = readExitFile()
+            //Use the status from the exitStatus file if it exists. Else use the exit code from the pipeline operation
+            task.stdout = outputFile
+            task.exitStatus = xs != null ? xs :  operation.getError()?.getCode()
+            task.stderr = xs  != null ?  errorFile : operation.getError()?.getMessage()
+
+            /* If we get error 10 or 14 from the pipelines api we'll retry
+             * since it means that the vm instance was probably preemptied.
+             *
+             * This retry method is a bit crude, find a cleaner method if it is available
+             */
+            if(!xs && (task.exitStatus == 10 || task.exitStatus == 14)) {
+                task.config.setProperty("maxRetries",task.failCount+1)
+                task.config.setProperty("errorStrategy","retry")
+            }
+            status = TaskStatus.COMPLETED
+            return true
+        } else
+            return false
+    }
+
+    private List<Event> extractRuntimeDataFromOperation() {
+        def metadata = operation.getMetadata() as Metadata
+        if (!this.metadata) {
+            this.metadata = metadata
+            return metadata?.getEvents()
+        } else {
+            //Get the new events
+            def delta = metadata.getEvents().size() - this.metadata.getEvents().size()
+            this.metadata = metadata
+            return delta > 0 ? metadata.getEvents().take(delta) : [] as List<Event>
+        }
+    }
+
+    private Integer readExitFile() {
+        try {
+            exitFile.text as Integer
+        }
+        catch (Exception e) {
+            log.debug "[GOOGLE PIPELINE] Cannot read exitstatus for task: `$task.name`", e
+            null
+        }
+    }
+
+    @Override
+    void kill() {
+        log.debug "[GOOGLE PIPELINE] Killing pipeline '$operation.name'"
+        executor.helper.cancelOperation(operation)
+    }
+
+    @Override
+    void submit() {
+
+        final launcher = new GooglePipelinesScriptLauncher(this.taskBean, this)
+        launcher.build()
+
+        String stagingScript = """
+           mkdir -p $task.workDir ;
+           chmod 777 $task.workDir ;
+           ${stagingCommands.join(" ; ")} ;
+           cd $task.workDir ;
+           chmod 777 ${TaskRun.CMD_SCRIPT} ${TaskRun.CMD_RUN} ;
+           ls -haltr $mountPath             
+        """.stripIndent().leftTrim()
+
+        String mainScript = "cd $task.workDir ; echo \$(./${TaskRun.CMD_RUN}) | bash 2>&1 | tee ${TaskRun.CMD_LOG}"
+
```

Comment:
Not sure to understand the logic this command? 

```
$(./.command.run) | bash 2>&1 | tee .command.log
```  

Would not be easier to use the following?

```
.command.run 2>&1 | tee .command.log
```

Reply:
Hi.

I think it was a part of my attempt to avoid having to "chmod 777" the workdir by piping the command file into bash instead of executing it.  It didn't work but I seem to have, again, forgot to revert it to your suggestion.  It should be safe to change it to the easier form.

Reply:
Ok, I will give it a try